internal/configs/sharding/allowlist.go

```diff
@@ -0,0 +1,47 @@
+package shardingconfig
+
+import (
+	"fmt"
+
+	bls_cosi "github.com/harmony-one/harmony/crypto/bls"
+)
+
+type Allowlist struct {
+	MaxLimitPerShard int
+	BLSPublicKeys    []bls_cosi.PublicKeyWrapper
+}
+
+func _BLS(pubkeys []string) []bls_cosi.PublicKeyWrapper {
+	blsPubkeys := make([]bls_cosi.PublicKeyWrapper, len(pubkeys))
+	for i := range pubkeys {
+		if key, err := bls_cosi.WrapperPublicKeyFromString(pubkeys[i]); err != nil {
+			panic(fmt.Sprintf("invalid bls key: %d:%s error:%s", i, pubkeys[i], err.Error()))
+		} else {
+			blsPubkeys[i] = *key
+		}
+	}
+	return blsPubkeys
+}
+
+// each time to update the allowlist, it requires a hardfork.
```

Comment:
Is there a better way out of this? How frequently do you think we'd be updating the allow list?

Reply:
I don't think we should update the allow list frequently, maybe once a month?

Reply:
@sophoah Can you comment if this is a feasible strategy: a hard fork every month until the leader allow list is in effect?

Reply:
> Is there a better way out of this? How frequently do you think we'd be updating the allow list?

hardfork is required each time the list is updated. unless we put it into the stateDB and update it with transactions(via precompiled contract).

Reply:
Although we could write allowlist and its epoch to an external config file, it would still cause hardfork if it is updated. nodes use different config file may pick different leaders. and nodes start at early block must know the allowlist at that block.

Reply:
> s is a feasible strategy: a hard fork

to be honest I really don't like the idea of a new hardfork every time we update the list or doing that every month ... best is something that can allow the update of the list without doing a hardfork. 

ie maybe just skip or ignore a bls key if one that is the list doesn't answer. I think we are already doing that, 

The challenge based on my understand here is to have the same list in all validator node so things can flow in order .. that's the main reason of the hardfork for me

Reply:
once a month hardfork is not something we can really do ... hardfork is really difficult to coordinate.

Reply:
@rlan35 What do you think? Updating the allow list via a precompile should be less complicated to manage compared to hard forking for each allow list addition.

Reply:
The allowlist feature is only a temporary experiment to test out having external validators becoming leaders. The list shouldn't change frequently. Maybe at most 2-3 times before the true decentralized leader rotation is launched and this feature gets deprecated.